### PR TITLE
Fetch followed artists by album

### PR DESCRIPTION
**Pull request by hedgehog1833**
A new endpoint to fetch all followed artists by getting liked albums from Spotify. A list of `fetchTypes` is required to tell which way of importing is requested. I chose not to use enums because I didn't want to have to handle parsing errors here. If a type does not match, just nothing is returned. I duplicated the return type to avoid confusion with the other method. The other one will be removed sooner or later anyway.
Only those artists are returned that the user does not follow yet. This makes the list smaller if the import is triggered again later.

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Artist import endpoints](https://trello.com/c/ILFpbOQk/243-artist-import-endpoints)

File: webapp/src/main/java/rocks/metaldetector/service/spotify/SpotifyFollowedArtistsServiceImpl.java [hunk 1]
```diff
@@ -51,8 +57,37 @@
...
+        () -> new ResourceNotFoundException("User with public id '" + publicUserId + "' not found!")
+    );
+
+    Set<ArtistDto> artistDtos = new HashSet<>();
+
+    for (String fetchType : fetchTypes) {
+      if (fetchType.equalsIgnoreCase(ARTIST_FETCH_TYPE_ALBUM)) {
+        artistDtos.addAll(getArtistsFromLikedAlbums(currentUser));
+      }
+    }
```
Comment (DanielW1987) [inline webapp/src/main/java/rocks/metaldetector/service/spotify/SpotifyFollowedArtistsServiceImpl.java:73]: you can eliminate the for loop if you work with contains.

File: webapp/src/main/java/rocks/metaldetector/web/controller/rest/SpotifyRestController.java [hunk 2]
```diff
@@ -32,7 +37,14 @@
...
                produces = {MediaType.APPLICATION_XML_VALUE, MediaType.APPLICATION_JSON_VALUE})
   public ResponseEntity<SpotifyArtistImportResponse> importArtists() {
-    List<ArtistDto> artists = artistImportService.importArtistsFromLikedReleases();
+    List<ArtistDto> artists = spotifyFollowedArtistsService.importArtistsFromLikedReleases();
     return ResponseEntity.ok(new SpotifyArtistImportResponse(artists));
   }
+
+  @GetMapping(path = Endpoints.Rest.SPOTIFY_FOLLOWED_ARTISTS,
+              produces = {MediaType.APPLICATION_XML_VALUE, MediaType.APPLICATION_JSON_VALUE})
+  public ResponseEntity<SpotifyFollowedArtistsResponse> getFollowedArtists(@RequestParam(value = FETCH_TYPES_PARAM) @NotEmpty List<String> fetchTypes) {
```
Comment (DanielW1987) [inline webapp/src/main/java/rocks/metaldetector/web/controller/rest/SpotifyRestController.java:46]: Maybe you can the Enum use here and Spring will handle the Conversion.

Comment (hedgehog1833) [reply webapp/src/main/java/rocks/metaldetector/web/controller/rest/SpotifyRestController.java:46]: Oh yeah nice that works. For any other than the possible fetch types a bad request ist returned

**Review by DanielW1987 — APPROVED** [review]

**Pull request merged**